azure/services/asogroups/groups.go

```diff
@@ -90,39 +88,33 @@ func (s *Service) Delete(ctx context.Context) error {
 		return nil
 	}
 
-	// check that the resource group is not BYO.
-	managed, err := s.IsManaged(ctx)
-	if err != nil {
-		if apierrors.IsNotFound(err) {
-			// already deleted or doesn't exist, cleanup status and return.
-			s.Scope.UpdateDeleteStatus(infrav1.ResourceGroupReadyCondition, ServiceName, nil)
-			return nil
-		}
-		return errors.Wrap(err, "could not get resource group management state")
-	}
-	if !managed {
-		log.V(2).Info("Skipping resource group deletion in unmanaged mode")
-		return nil
-	}
-
-	err = s.DeleteResource(ctx, groupSpec, ServiceName)
+	err := s.DeleteResource(ctx, groupSpec, ServiceName)
```

Comment:
I think we're very close to completely getting rid of this code and having one common function for all services (food for thought)

Reply:
Yep, I've been trying to work in that direction where it seems reasonable for this first pass.